Python(Go) - pipenv - Not able to generate SPDX file

@lfpratik 
**Tool Version** v0.0.6
**Test Repo** https://github.com/lfpratik/spdx-pipenv-demo
**OS** Windows 10

1. Followed all prerequisite steps as per https://confluence.linuxfoundation.org/display/PROD/SPDX+-+Python+Module+-+Prerequisites+For+Windows
2. Followed Prerequisite and Steps as per below screenshot
![image](https://user-images.githubusercontent.com/73597479/121533821-462b3b00-ca1e-11eb-8868-09f28bed00a6.png)
3. Execute ./spdx-sbom-generator
4. Observed that SPDX file is not generated and below error is displayed

```
ASUS@LAPTOP-2QB11PTS MINGW64 /d/LFX/Projects/SPDX/Python/Go/spdx-pipenv-demo (main)
$ pipenv sync
Installing dependencies from Pipfile.lock (5a4c19)...
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
All dependencies are now up-to-date!

ASUS@LAPTOP-2QB11PTS MINGW64 /d/LFX/Projects/SPDX/Python/Go/spdx-pipenv-demo (main)
$ pipenv install
Installing dependencies from Pipfile.lock (5a4c19)...
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.

ASUS@LAPTOP-2QB11PTS MINGW64 /d/LFX/Projects/SPDX/Python/Go/spdx-pipenv-demo (main)
$ spdx-sbom-generator
bash: spdx-sbom-generator: command not found

ASUS@LAPTOP-2QB11PTS MINGW64 /d/LFX/Projects/SPDX/Python/Go/spdx-pipenv-demo (main)
$ ./spdx-sbom-generator
INFO[2021-06-10T19:07:23+05:30] Starting to generate SPDX ...
INFO[2021-06-10T19:07:23+05:30] Running generator for Module Manager: `pip` with output `bom-pip.spdx`
       21-06-10T19:07:24+05:30] Current Language Version Python 3.9.5
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x30 pc=0x7940cd]

goroutine 1 [running]:
spdx-sbom-generator/internal/models.(*CheckSum).String(0x0, 0x0, 0x0)
        /github/workspace/internal/models/models.go:76 +0x2d
spdx-sbom-generator/internal/format.(*Format).convertToPackage(0xc002eb9b08, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /github/workspace/internal/format/format.go:180 +0xd0
spdx-sbom-generator/internal/format.(*Format).buildPackages(0xc002eb9b08, 0xc000075800, 0x33, 0x34, 0xc002fbea86, 0x13, 0xc0034d33e0, 0x2a, 0xc003333ab0, 0x61)
        /github/workspace/internal/format/format.go:153 +0x125
spdx-sbom-generator/internal/format.(*Format).Render(0xc002eb9b08, 0xc0030144e0, 0x0)
        /github/workspace/internal/format/format.go:59 +0x13b
spdx-sbom-generator/internal/handler.(*spdxHandler).Run(0xc002ee6240, 0x6, 0xd311d5)
        /github/workspace/internal/handler/spdx.go:89 +0x548
main.generate(0x1644880, 0x166c0e8, 0x0, 0x0)
        /github/workspace/cmd/generator/generator.go:105 +0x43a
github.com/spf13/cobra.(*Command).execute(0x1644880, 0xc0000581a0, 0x0, 0x0, 0x1644880, 0xc0000581a0)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x2b1
github.com/spf13/cobra.(*Command).ExecuteC(0x1644880, 0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
        /github/workspace/cmd/generator/generator.go:37 +0x64
```

--- a/internal/modules/pip/worker/decoder.go
+++ b/internal/modules/pip/worker/decoder.go
@@ -80,8 +80,7 @@ func (d *MetadataDecoder) BuildMetadata(pkg Packages) Metadata {
 
 	metadata.Root = pkg.Root
 	metadata.ProjectURL = BuildProjectUrl(metadata.Name, metadata.Version)
-	metadata.PackageURL = BuildPackageUrl(metadata.Name, metadata.Version)
-	if len(metadata.HomePage) > 0 {
+	if len(metadata.HomePage) > 0 && metadata.HomePage != "None" {
 		metadata.PackageURL = metadata.HomePage
 	}
 	metadata.PackageJsonURL = BuildPackageJsonUrl(metadata.Name, metadata.Version)

--- a/internal/modules/pip/worker/pypi.go
+++ b/internal/modules/pip/worker/pypi.go
@@ -88,11 +88,11 @@ var HashAlgoPickOrder []models.HashAlgorithm = []models.HashAlgorithm{
 func getPypiPackageChecksum(packagename string, packageJsonURL string, checkfortag bool, wheeltag string) models.CheckSum {
 	checksum := models.CheckSum{
 		Algorithm: models.HashAlgoSHA1,
-		Value:     "",
 	}
+
 	packagedigests, err := fetchAndDecodePypiPackageDataJSON(packagename, packageJsonURL)
 	if err != nil {
-		checksum.Value = NoAssertion
+		checksum.Content = []byte(packagename)
 		return checksum
 	}
 
@@ -115,7 +115,6 @@ func getPypiPackageChecksum(packagename string, packageJsonURL string, checkfort
 		return checksum
 	}
 
-	checksum.Value = NoAssertion
 	return checksum
 }
 
@@ -134,7 +133,7 @@ func getHighestHashData(packagedigests PackageDigest) (models.HashAlgorithm, str
 		}
 	}
 
-	return models.HashAlgoSHA1, NoAssertion
+	return algoType, digestValue
 }
 
 func getPackageTypeDigestBDist(packagedigests *[]PackageDigest, wheeltag string) (PackageDigest, error) {

